tensorflow_io/bigquery/kernels/bigquery_lib.h

```diff
@@ -50,17 +52,46 @@ Status GetDataFormat(string data_format_str,
 class BigQueryClientResource : public ResourceBase {
  public:
   explicit BigQueryClientResource(
-      std::shared_ptr<apiv1beta1::BigQueryStorage::Stub> stub)
-      : stub_(std::move(stub)) {}
-
-  std::shared_ptr<apiv1beta1::BigQueryStorage::Stub> get_stub() {
-    return stub_;
+      std::function<std::unique_ptr<apiv1beta1::BigQueryStorage::Stub>(
+          const string &read_stream)>
+          stub_factory)
+      : stub_factory_(stub_factory) {}
+
+  explicit BigQueryClientResource()
+      : BigQueryClientResource([](const string &read_stream) {
+          string server_name = "dns:///bigquerystorage.googleapis.com";
+          auto creds = ::grpc::GoogleDefaultCredentials();
+          grpc::ChannelArguments args;
+          args.SetMaxReceiveMessageSize(kMaxReceiveMessageSize);
+          args.SetUserAgentPrefix(
+              strings::StrCat("tensorflow-", TF_VERSION_STRING));
+          args.SetInt(GRPC_ARG_KEEPALIVE_PERMIT_WITHOUT_CALLS, 0);
+          args.SetInt(GRPC_ARG_KEEPALIVE_TIMEOUT_MS, 60 * 1000);
+          // To prevent gRPC from reusing channel
+          args.SetString("read_stream", read_stream);
```

Comment:
you might want to hash % some_number here in case the number of streams returned is large.

Reply:
read_stream uniquely identifies BQ Storage API stream and # of streams can't be too large, usually it is less than 10.
Besides if those custom args are persisted somewhere in gRPC logs, I'd rather have it as a stream name instead of some integer